CRM/Utils/System/Drupal8.php

```diff
@@ -571,7 +571,9 @@ public function getModules() {
     $module_data = \Drupal::service('extension.list.module')->reset()->getList();
     foreach ($module_data as $module_name => $extension) {
       if (!isset($extension->info['hidden']) && $extension->origin != 'core' && $extension->status == 1) {
-        $modules[] = new CRM_Core_Module('drupal.' . $module_name, TRUE);
+        $modules[] = new CRM_Core_Module('drupal.' . $module_name, TRUE,
+          ts('%1 (%2)', [1 => $extension->info['name'] ?? $module_name, 'Drupal'])
```

Comment:
I'm not really following the general issue but how does this `ts` work?

i.e. how can the translator translate `'%1 (%2)'`? And why the second parameter "Drupal" which is a constant?

Reply:
Hmm, @totten maybe use the internal `_ts()` so as not to confuse translators?

Reply:
> (@demeritcowboy) i.e. how can the translator translate '%1 (%2)'?

Isn't this one of those strings where it's yucky no matter what you do?

* `ts('%1 (%2)')` (*no meaningful hint for translator...*)
* `sprintf('%s (%s)')` (*weird on RTL languages...*)

AFAIK, making it sensible would require structural change to string-translation, e.g.

* `ts('%module (%platform)', ...)`  (*named parameters rather than numeric parameters...*)
* `ts('MODULE_PLATFORM_LABEL', ...)` (*symbolic string identifiers rather than template-expressions...*)

That's obviously a different/bigger project. I'm fine with switching it to `sprintf()` or `_ts()` (or keeping `ts()`).

> (@demeritcowboy) why the second parameter "Drupal" which is a constant?

Well, in the `DrupalBase.php` variant (d5453d5; D7/BD), I had used the same string -- but with`$params[2]` assigned as `CIVICRM_UF` ("Backdrop" or "Drupal"). For the `Drupal8.php` variant (3e939a4), the literal `Drupal` seemed better than  `CIVICRM_UF` (*since that would render as `Drupal8` when running on D9 or D10*).

FWIW, I did assume that the names "Drupal" and "Backdrop" would be rendered the same in other languages. I just tested that theory with Google Translate. (*Translate "Jack installed the Drupal application" to a dozen languages.*) All the languages with Latin-like alphabets used "Drupal". For non-Latin alphabets, Google was mixed -- most used the Latin "Drupal" (*eg Chinese, Greek, Hindi, Japanese, Punjabi, Russian, Thai, Ukrainian*), but several used transliterations (*eg Arabic, Armenian, Hebrew, Bengali*). I don't see any rhyme-or-reason to that variation.

Of course, that could be a limitation in Google Translate.  Probably the best interpretation is: "In online technical discourse, application-names may or may not be transliterated, depending on `$ARBITRARY`."

> (@colemanw) maybe use the internal `_ts()`

Thanks, I had forgotten about that option. `_ts(CIVICRM_UF)` seems better than `CIVICRM_UF` -- it would allow the possibility of using the transliterated name.

Reply:
@demeritcowboy @colemanw - OK, I switched both levels (the string-arrangement and the UF name) to use `_ts()`. This is a sort of non-committal compromise:

* If these strings (`Drupal`, `Backdrop`, `%1 (%2)`) have been translated previously (*or are translated in the future*), then this will follow the translated form.
* This patch won't add new tasks to the Transifex workflow -- which is fine, since these strings should look the same for 98%+ of Civi sites.